MatrixSDK/MXSession.m

```diff
@@ -1538,14 +1545,12 @@ - (void)handleServerSyncError:(NSError*)error forRequestWithServerTimeout:(NSUIn
         {
             MXLogDebug(@"[MXSession] The connection has been cancelled in state %@", @(_state));
 
-            if (_state == MXSessionStateSyncInProgress)
+            if (self.isPauseable)
             {
                 // This happens when the SDK cannot make any more requests because the app is in background
                 // and the background task is expired or going to expire.
                 // The app should have paused the SDK before but it did not. So, pause the SDK ourselves.
-                // Note that we need to come back to MXSessionStatePauseRequested in order to be able to pause.
                 MXLogDebug(@"[MXSession] -> Go to pause");
-                [self setState:MXSessionStatePauseRequested];
```

Comment:
Just a note to indicate that I double checked this change and it looks sane to me.